### ISSUE TITLE
Add modulation for square waves?

### ISSUE THREAD

**Issue by dxxb** [moduscreate/ATMlib#6]
After swapping the inaudible triangle waveform we are left with 3 square waves (25,50,75) duty cycle respectively. The following change to the synth if applied to all 3 oscillators would give us per channel duty cycle modulation control... 😉 The change is cheap in terms of cycles and will make us even with the cycles we reaped by swapping triangle with square wave. Do we want it?

``` diff
@@ -145,11 +146,13 @@ ISR(TIMER4_OVF_vect, ISR_NAKED)
 "      sts  osc+0*%[osz]+%[pha]+1, r1                   " ASM_EOL
 "; OSC 0 pulse waveform (25/75 duty-cycle square)    " ASM_EOL
-"      mov  r18,                   r1                   " ASM_EOL
-"      lsl  r18                                         " ASM_EOL
-"      and  r18,                   r1                   " ASM_EOL
+"      lds  r27,                   osc+0*%[osz]+%[mod]  " ASM_EOL
+"      cp   r1,                    r27                  " ASM_EOL
 "      lds  r27,                   osc+0*%[osz]+%[vol]  " ASM_EOL
-"      sbrc r18,                   7                    " ASM_EOL
+"      brcs skip_neg                                    " ASM_EOL
 "      neg  r27                                         " ASM_EOL
+"skip_neg:                                           " ASM_EOL
 "      add  r26,                   r27                  " ASM_EOL
 "; OSC 1 advance phase accumulator                   " ASM_EOL
 "      lds  r18,                   osc+1*%[osz]+%[phi]  " ASM_EOL
```

**Comment by jaygarcia** [comment]
Yes please!!! 🎶

**Comment by dxxb** [comment]
Implemented in 4fc4cc38e1b8f3d6b053f7f0c12dddde96098357, later commits added set and slide support for duty cycle. Once I get around refactoring vibrato and tremolo FX I will add support for duty cycle to the vibrato/tremolo _LFO_ (FYI the existing _LFO_ uses a triangular waveform right now to avoid storing a sine table).
